src/opencl_common.c

```diff
@@ -1536,8 +1536,21 @@ static cl_ulong gws_test(size_t gws, unsigned int rounds, int sequential_id)
 	self->methods.clear_keys();
 	{
 		char key[PLAINTEXT_BUFFER_SIZE];
-		int len = MAX(MIN(self->params.plaintext_length, 7),
-		              self->params.plaintext_min_length);
+		int len;
+
+		if (mask_add_len > 0 &&
+		    (options.req_minlength == -1 && options.req_maxlength == 0)) {
+			//-mask and no min/max are set
+			len = mask_add_len;
+		} else {
+			len = (self->params.benchmark_length & 0x7f);
+			len = (len < options.req_minlength) ? options.req_minlength: len;
+			len = (options.req_maxlength && len > options.req_maxlength) ?
+			          options.req_maxlength: len;
```

Comment:
Nitpick: I think `len = cond ? something : len` (note `len` used on both sides) is misuse of the ternary operator, and would be cleaner written using `if`. So instead of:

```c
len = (len < options.req_minlength) ? options.req_minlength: len;
len = (options.req_maxlength && len > options.req_maxlength) ? options.req_maxlength: len;
```

We can have:

```c
if (len < options.req_minlength)
        len = options.req_minlength;
if (options.req_maxlength && len > options.req_maxlength)
        len = options.req_maxlength;
```